keys/keys.go

```diff
@@ -0,0 +1,149 @@
+package keys
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	"github.com/google/uuid"
+)
+
+type (
+	// Key represents a kafka key.
+	Key interface {
+		sarama.Encoder
+	}
+
+	// CreateTimer is a create timer or delete timer event, value should be a messaging.CreateTimer or nil.
+	CreateTimer struct {
```

Comment:
I am confused why you need two different struct types for CreateTimer and ExecuteTimer - they contain the same fields? Why can you not just have a single struct called key?

Reply:
Oh I see why you did this now. How about instead we define a struct called key and then we can have two types of key CreateTimerKey and ExecuteTimerKey?

Reply:
I like the idea of doing this in protos!